src/ORM/Connect/MySQLiConnector.php

```diff
@@ -81,15 +81,20 @@ public function connect($parameters, $selectDB = false)
         // Connection charset and collation
         $connCharset = Config::inst()->get(MySQLDatabase::class, 'connection_charset');
         $connCollation = Config::inst()->get(MySQLDatabase::class, 'connection_collation');
+        $socket = static::config()->get('socket');
+        $flags = static::config()->get('flags');
```

Comment:
It seems weird to use config for this when everything else about a database connection is using env vars. Can you please explain your thinking here?

Reply:
All the other db specific values `ssl_cipher_default`, `connection_charset`  are config values so lets keep consistent

Reply:
I feel like the socket you're using to connect has more to do with the environment-specific infrastructure, while the charset your connection uses is more to do with how the data is stored.

i.e. you want the data to be consistent regardless of what you're connecting to - but each environment may want to connect to a database on a different port.

Reply:
Yeah either or. Though you can always set config based on env vars so config gives a bit more flex.

Reply:
I would prefer for everything to do with _what_ database you're connecting to to be consistent. For CMS 5 that means explicitly using ENV vars.

We can change it in CMS 6 to use config which defaults to those env vars, if you want config.

Reply:
Sure so `SS_DATABASE_SOCKET` How do you imagine the `SS_DATABASE_FLAGS` to work? I don't think it would function as a string.

Reply:
We could either introduce some functionality to allow array values from env vars, or more simply, just use a comma (or other character) delimited string. Split by the delimiter, use `constant()` like you're doing below to get the specific flags based on the names.

Reply:
Updated to env vars